src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -931,6 +1008,49 @@ case class TableImport(paths: Array[String], typ: TableType, readerOpts: TableRe
   }
 }
 
+case class TableRange(n: Int, name: String = "index", nPartitions: Int) extends TableIR {
```

Comment:
Can you have optional parameters before non-optional in Scala?

Reply:
Yes!

```
scala> def f(i: Int, j: Int = 5, k: Int): Int = i + j + k
f: (i: Int, j: Int, k: Int)Int

scala> f(1, 2, 3)
res0: Int = 6

scala> f(1, k=5)
res3: Int = 11
```

However, I admit it is a bit strange and it isn't used, so I'll remove it.